internal/orchestrator/status.go

```diff
@@ -129,3 +133,123 @@ func (or *orchestrator) GetStatus(ctx context.Context) (status *core.NamespaceSt
 
 	return status, nil
 }
+
+// Get the earliest incomplete identity claim message for this org, if it exists
+func (or *orchestrator) getRegistrationMessage(ctx context.Context) (msg *core.MessageInOut, err error) {
+	fb := database.MessageQueryFactory.NewFilter(ctx)
+	filter := fb.Sort("created").Limit(1).And(
+		fb.Eq("type", core.MessageTypeDefinition),
+		fb.Eq("tag", core.SystemTagIdentityClaim),
+		fb.Eq("author", core.FireFlyOrgDIDPrefix+or.config.Multiparty.Org.Name),
+		fb.In("state", []driver.Value{core.MessageStateStaged, core.MessageStateReady, core.MessageStateSent, core.MessageStatePending}),
+	)
+	msgs, _, err := or.GetMessagesWithData(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	if len(msgs) > 0 {
+		return msgs[0], nil
+	}
+	return nil, nil
+}
+
+// Ensure the given registration message correlates with the expected type
+func (or *orchestrator) checkRegistrationType(ctx context.Context, msg *core.MessageInOut, expectedType core.IdentityType) (match bool, err error) {
```

Comment:
Was this check in response to a specific error scenario we've seen? I'm trying to think why there would be a pending registration message, but the contents would be inconsistent.

Reply:
No, it was purely theoretical - it felt clumsy before I added this that we would find a registration message and just _assume_ it was of the correct type. It's technically possible that a pending node registration message could be there without the org being registered, and the API would then return the wrong response.

I admit it could be overkill...